src/main.js

```diff
@@ -1,83 +1,83 @@
-//     keyword: "rhino",
-//     horns: 2,
-//   },
-
-//   {
-//     image_url:
-//       "https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg",
-//     title: "Unicorn Head",
-//     description: "Someone wearing a creepy unicorn head mask",
-//     keyword: "unicorn",
-//     horns: 1,
-//   },
-// ];
-
-// function displayBeasts(array) {
-//   for (let beast of array) {
-//     let img = this.image_url;
-//     let alt = "horned beast image";
-//     let title = this.title;
-//     let disc = this.description;
-
-//     return `<HornedBeasts imgSrc="${img}" alt="${alt} title="${title} />`
-//     console.log("beasts func is funcin")
-//   }
-// }
-
-// let singleVariable = displayBeasts(beastsArray);
-// console.log(singleVariable)
+import beastData from './data.json';
+import Image from "react-bootstrap/Image";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Container from "react-bootstrap/Container";
 
 class Main extends Component {
   
   render() {
     return (
-      <>
+      <Container>
         <h1>Main is maining</h1>
+        <Row>
+          <Col>
         <HornedBeasts
-          imgSrc="https://place-hold.it/300x500"
-          alt="Horned Image"
-          title="idk horned title"
+          imgSrc={beastData[0].image_url}
+          alt={beastData[0].keyword}
+          title={beastData[0].title}
+          description={beastData[0].description}
         />
+          </Col>
+          <Col>
         <HornedBeasts
-          imgSrc="https://place-hold.it/300x500"
-          alt="Horned Image"
-          title="idk horned title"
+          imgSrc={beastData[1].image_url}
+          alt={beastData[1].keyword}
+          title={beastData[1].title}
+          description={beastData[1].description}
         />
-      </>
+          </Col>
+          <Col>
+        <HornedBeasts
+          imgSrc={beastData[2].image_url}
+          alt={beastData[2].keyword}
+          title={beastData[2].title}
+          description={beastData[2].description}
+        />
+          </Col>
+        </Row>
+        <Row>
+          <Col>
+        <HornedBeasts
+          imgSrc={beastData[3].image_url}
+          alt={beastData[3].keyword}
+          title={beastData[3].title}
+          description={beastData[3].description}
+        />
+          </Col>
+          <Col>
+          </Col>
+          <Col>
+          </Col>
+        </Row>
+      </Container>
     );
   }
 }
 
 class HornedBeasts extends Component {
 
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      favorites: 0
+    }
+  }
+
   render() {
     return (
       <>
-        <h2>Horned title</h2>
-        <img
+        <h2>{this.props.title}</h2>
+        <Image
           src={this.props.imgSrc}
           alt={this.props.alt}
           title={this.props.title}
+          onClick={() => this.setState({ favorites: this.state.favorites + 1 })}
```

Comment:
Nice work so far:

Feedback: Move this setState call out of the render as this can cause some weird things to happen.

Instead, create a separate method (called handleClickedBeasts (or something, even handleClick etc.). Inside, do your setState call and then pass the click handler to your onClick prop in the render.